### DISCUSSION TITLE
Problem importing swup from node_modules

### DISCUSSION THREAD

**Discussion by danbraun** [swup/swup#523] [Help] [answered]
Seems that swup though found, is not loading. Any help?

Using:
`import Swup from 'swup';
const swup = new Swup();`

seeing this error (and swup is not working)

`Could not find a declaration file for module 'swup'. 'd:/Sites/parcel-pineapple/node_modules/swup/lib/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/swup` if it exists or add a new declaration (.d.ts) file containing `declare module 'swup';`ts(7016)`

**Comment by daun** [comment, marked as answer]
The error you're seeing is a TS error, that's to be excepted since swup currently isn't written in JS.

Can you share the output of your bundler? If there's an issue loading from node_modules, it should show up there.

**Comment by danbraun** [comment]
Thank you for your patient help. I believe I was expecting this to work without a bundler. I'm now running Parcel, and with script type='module' in my html source. Swup works now. Though that _warning_ still displays when I hover over the import in my source js file, I understand that it is irrelevant and can ignore it.

**Reply by daun** [reply]
Good to know! FYI, the upcoming 3.0 version of swup will have an ESM export so you will be able to use it without a bundler.
